qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -330,6 +330,51 @@ def duration(self) -> float:
 
         return np.sum(self.durations)
 
+    def resample(self, time_step: float, name: Optional[str] = None) -> "DrivenControl":
+        r"""
+        Returns a new driven control obtained by resampling this control.
+
+        Parameters
+        ----------
+        time_step : float
+            The time step to use for resampling, :math:`\delta t`.
+        name : str, optional
+            The name for the new control. Defaults to ``None``.
+
+        Returns
+        -------
+        DrivenControl
+            A new driven control, sampled at the specified rate. The durations of the new control
+            are all equal to :math:`\delta t`. The total duration of the new control might be
+            slightly larger than the original duration, if the time step doesn't exactly divide the
+            original duration.
+        """
+        check_arguments(
+            time_step > 0,
+            "Time step must be positive.",
+            {"time_step": time_step},
+        )
+        check_arguments(
+            time_step <= self.duration,
```

Comment:
What if the `time_step` is larger than the minimum segment duration time, but smaller than duration? I guess we don't expect it?

Reply:
Yeah, I was thinking that's just a case we need to accept. Even for a sensible `time_step`, you can still get a wonky sampling if it doesn't exactly divide a segment duration, which I think is unavoidable.